plugin/qmcedeclarativeplugin.cpp

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2019 Jolla Ltd.
+ * Copyright (c) 2016 - 2022 Jolla Ltd.
```

Comment:
I think this should be `2016-2019, 2022`. Only years during which changes were actually made should be listed. Think it from the other side, imagine OMP would do a change in 2022 and changed their notice to `2019-2022`, which would falsely indicate they did contribute in 2019 and 2022 and every year between. More occurrences elsewhere.

Reply:
I think we've commonly just had the start and end roughly everywhere. Trying to list all years that had some changes shouldn't bring much benefit, just clutter the header.

Reply:
Let's take a look at what FSF says on this, as there are *some* source codes in our ecosystem that need to follow their rules, so it makes sense to use these as a base.

> To update the list of year numbers, add each year in which you have made nontrivial changes to the package.
>
> When you add the new year, it is not required to keep track of which files have seen significant changes in the new year and which have not. It is recommended and simpler to add the new year to all files in the package, and be done with it for the rest of the year. 
>
> You can use a range (‘2008-2010’) instead of listing individual years (‘2008, 2009, 2010’) if and only if: 1) every year in the range, inclusive, really is a “copyrightable” year that would be listed individually; and 2) you make an explicit statement in a README file about this usage.

https://www.gnu.org/prep/maintain/html_node/Copyright-Notices.html

So no need to maintain years per-file as we have been doing commonly - our bad. But illegal to list years that are not "copyrightable" and I have a feeling there were some in case of this package.

Reply:
> Let's take a look at what FSF says on this, as there are some source codes in our ecosystem that need to follow their rules, so it makes sense to use these as a base.

FSF should mostly matter if we make upstream contributions that adjust copyright years. Which shouldn't be too often.

> So no need to maintain years per-file as we have been doing commonly - our bad. But illegal to list years that are not "copyrightable" and I have a feeling there were some in case of this package.

Illegal on what jurisdiction? By law or by court precedent?

To me this is mostly just a convention to list the years. Practically mattering maybe when the copyright expires then some day when we're all dead. And even then should cover the entire range to freely use the file.

From FSF docs could quote also "When you add the new year, it is not required to keep track of which files have seen significant changes in the new year and which have not. It is recommended and simpler to add the new year to all files in the package, and be done with it for the rest of the year."

Reply:
> FSF should mostly matter if we make upstream contributions that adjust copyright years. Which shouldn't be too often.

What I quoted is a FSF's recommendation on "how to apply copyrights", not "how to not break copyrights".

> Illegal on what jurisdiction? By law or by court precedent? [...]

o_O

> From FSF docs could quote also [...]

That's essentially what I have quoted before, so..?

So let's summarize: I pointed out there is a breach to a commonly used convention that would take about a minute to fix and keep aligned with the rest of the world. Instead of that, two programmers spent more than half an hour on nonsense arguing whether the commonly used convention is worth to follow.